### PR TITLE
docs: Migrate /docs items to website 

### LINKED ISSUE #824: [DOC] Create reference website section
### Change in the documentation

Create a "Reference" section on the website which should hold all of the more "technical" subjects as subpages / sections.

Move `docs/configuration/configuration.md` and docs/configuration/flagd_start.md` to website

### LINKED ISSUE #826: [DOC] Move OpenTelemetry page to /reference
### Change in the documentation

_No response_

### LINKED ISSUE #827: [DOC] Move /docs/configuration/sem_ver_evaluation.md to website
### Change in the documentation

Move `docs/configuration/sem_ver_evaluation.md` to website

### PR DESCRIPTION
**Pull request by agardnerIT**
## This PR
- Creates `reference` menu item on website
- Migrates `docs/configuration/configuration.md` to website
- Migrates `docs/configuration/flagd.md` to website
- Migrates `docs/configuration/flagd_start.md` to website

### Related Issues
- Closes #824 
- Closes #826 
- Closes #827

### CODE REVIEW

**Comment by agardnerIT** [comment]
this is good to go. Don't know why the `diff` is failing

**Comment by agardnerIT** [comment]
Some lessons learned here:

1. `make generate-docs` fails on windows:

```
make generate-docs
FIND: Parameter format not correct
cd flagd; go run ./cmd/doc/main.go
The system cannot find the path specified.
make: *** [Makefile:74: generate-docs] Error 1
```

2. Changing dir into `/flagd` then running the go command with go < 1.20.6 failed with an Exception in `runtime.cgocall`.
    I needed to update to `golang 1.21.0` and it worked

3. The commands alter my files in a way that I do not want. I don't want that boilerplate as we're moving away from `/docs` so I **do** want a single line pointing people to the website.

```
git diff ../docs/configuration/flagd.md
diff --git a/docs/configuration/flagd.md b/docs/configuration/flagd.md
index bcfa899..97c4ae6 100644
--- a/docs/configuration/flagd.md
+++ b/docs/configuration/flagd.md
@@ -1,3 +1,18 @@
-# flagd
+<!-- markdownlint-disable-file -->
+## flagd
+
+Flagd is a simple command line tool for fetching and presenting feature flags to services. It is designed to conform to Open Feature schema for flag definitions.
+
+### Options
+
+```
+      --config string   config file (default is $HOME/.agent.yaml)
+  -x, --debug           verbose logging
+  -h, --help            help for flagd
+```
+
+### SEE ALSO
+
+* [flagd start](flagd_start)    - Start flagd
+* [flagd version](flagd_version)        - Print the version number of FlagD

-[https://flagd.dev](https://flagd.dev) is the new home of flagd documentation.
```

Long story short, I think we need to revisit the conditions applied in the PR workflow that enforce these checks for the new website structure.

Second, whatever is decided to be necessary could probably just be executed on the Action runner to remove this step from the user.

**Comment by agardnerIT** [comment]
@Kavindu-Dodan do you have any ideas on how to resolve the above to get this merged? Thanks!

**Comment by Kavindu-Dodan** [comment]
> @Kavindu-Dodan do you have any ideas on how to resolve the above to get this merged? Thanks!

I resolved the conflict and it should be good to go

**Comment by beeme1mr** [comment]
@Kavindu-Dodan can't we just change the location of the generated doc?

https://github.com/open-feature/flagd/blob/main/flagd/cmd/doc/main.go#L9

**Comment by agardnerIT** [comment]
This should now be all good and ready for a re-review (apart from the `make generate-docs`) failure above.

File: CONTRIBUTING.md
Comment (beeme1mr) [inline CONTRIBUTING.md:1]: I don't think this file should be moved to the website. A CONTRIBUTING file has special meeting in GitHub and anyone contributing would expect the content to live there.

File: web-docs/concepts/index.md [hunk 1]
```diff
@@ -213,7 +213,7 @@ The example above shows the `in` keyword being used, but flagd is also compatibl
 
 - [starts_with](https://github.com/open-feature/flagd/blob/main/docs/configuration/string_comparison_evaluation.md#startswith-evaluation-configuration)
 - [ends_with](https://github.com/open-feature/flagd/blob/main/docs/configuration/string_comparison_evaluation.md#endswith-evaluation-configuration)
```
Comment (beeme1mr) [inline web-docs/concepts/index.md:215]: Can these also be updated to a relative path?

File: web-docs/reference/flag_configuration.md [hunk 2]
```diff
@@ -0,0 +1,290 @@
...
+      "defaultVariant": "red",
+      "state": "ENABLED",
+      "targeting": {
+        "if": [
+          {
+            "$ref": "emailWithFaas"
+          },
+          {
+            "fractionalEvaluation": [
+              "email",
```
Comment (beeme1mr) [inline web-docs/reference/flag_configuration.md:263]:
```suggestion
            "fractional": [
              { "var": "email" },
```

**Comment by beeme1mr** [comment]
Duplicate of #927
